Libraries/Interaction/InteractionManager.js

```diff
@@ -73,13 +73,14 @@ var InteractionManager = {
   /**
    * Schedule a function to run after all interactions have completed.
    */
-  runAfterInteractions(callback: Function) {
-    invariant(
-      typeof callback === 'function',
-      'Must specify a function to schedule.'
-    );
-    scheduleUpdate();
-    _queue.push(callback);
+  runAfterInteractions(callback: ?Function): Promise {
+    return new Promise((resolve, reject) => {
+      scheduleUpdate();
+      if (callback) {
+        _queue.push(callback);
+      }
+      _queue.push(resolve);
+    });
```

Comment:
The callback is pushed onto the queue i.e. invoked when the interaction completes. That's my reading of @philikon's code at least. The InteractionManager wraps each callback in an ErrorUtils guard so even if the callback fails the InteractionManager will run the promise's resolve function.